File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maximum_profit = 0
        sum = 0
        for i in range(len(prices)-1):
            if(sum+(prices[i+1]-prices[i]) > 0):
                sum += (prices[i+1]-prices[i])
            elif((prices[i+1]-prices[i]) < 0):
                sum = 0
            else:
                sum = (prices[i+1]-prices[i])
            if(sum > maximum_profit):
                maximum_profit = sum
        return maximum_profit


        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int sum = 0;
        for(int i=0;i<prices.length-1;i++)
        {
            if(sum + (prices[i+1] - prices[i]) > 0)
            {
                sum = sum + (prices[i+1] - prices[i]);
            }
            else if ((prices[i+1] - prices[i]) < 0)
            {
                sum = 0;
            }
            else 
            {
                sum = (prices[i+1] - prices[i]);
            }
            if(sum > maxProfit)
            {
                maxProfit = sum;
            }
        }
        return maxProfit;
        
    }
}